libs/langchain/langchain/tools/graphql/tool.py

```diff
@@ -27,9 +32,13 @@ class Config:
         arbitrary_types_allowed = True
 
     def _run(
-        self,
-        tool_input: str,
-        run_manager: Optional[CallbackManagerForToolRun] = None,
+            self,
+            query: str,
+            query_variables: Optional[Dict[str, Any]] = None,
+            graphql_endpoint: Optional[str] = None,
+            headers: Optional[Dict[str, str]] = None,
```

Comment:
In general, we do not want the agent / LLM responsible for filling in information like headers which may have auth info